In [1]:
import pandas as pd,numpy as np,matplotlib.pyplot as plt,seaborn as sns
%matplotlib inline

In [3]:
WH=pd.read_excel('Working Hours.xlsx')
WH.head(3)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    NaN    NaN     NaN   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    NaN    NaN     NaN   
2 2022-04-12   9:30 am- 10:30 am    1.0    1.0    NaN    NaN     NaN   

  Mode of class  
0        Online  
1        Online  
2        Online

In [4]:
WH.isna().sum()

Date              0
Time              0
Jikky            51
Kayis            72
Sinny            68
Criji            74
Johnan           82
Mode of class     0
dtype: int64

# Data Cleaning

Nan values are replaced by zeros

In [6]:
WH=WH.fillna(0)
WH.head(3)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   
2 2022-04-12   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  
0        Online  
1        Online  
2        Online

In [7]:
WH.shape

(91, 8)

In [8]:
WH.columns

Index(['Date', 'Time', 'Jikky', 'Kayis', 'Sinny', 'Criji', 'Johnan',
       'Mode of class'],
      dtype='object')

In [9]:
WH.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91 entries, 0 to 90
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Date           91 non-null     datetime64[ns]
 1   Time           91 non-null     object        
 2   Jikky          91 non-null     float64       
 3   Kayis          91 non-null     float64       
 4   Sinny          91 non-null     float64       
 5   Criji          91 non-null     float64       
 6   Johnan         91 non-null     float64       
 7   Mode of class  91 non-null     object        
dtypes: datetime64[ns](1), float64(5), object(2)
memory usage: 5.8+ KB


## Splitting date

date variable is needed to be splitted for further analysis

here date is splitted into, month, year and day

In [10]:
WH['Month']=pd.to_datetime(WH['Date']).dt.month
WH['year']=pd.to_datetime(WH['Date']).dt.year
WH['Day']=pd.to_datetime(WH['Date']).dt.day
WH.head(3)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   
2 2022-04-12   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  
0        Online      4  2022   11  
1        Online      4  2022   11  
2        Online      4  2022   12

# Data Analysis

Data is analysed with these requirements

1.Find who had taken more class.

2.Find by which mode, most of the time class has taken in month June

3.Find the sum of Hours of each mentors

4.Find by which time hour most the class has taken

5.Find how many physical classes has taken by each mentors through out each month

6.if the the payment is 100 for one hour , then how much salary

7.if the the payment for one hour is 150 means how much salary did they got in each month

8.Create a dataframe using the data you got with salary of each month of each mentors

9.who earned more salary if the payment for one hour is 100.

10.who earned more salary if the payment for one hour is 150, in the month of june.

11.who is working more time

12.who had worked more hours in the month of May?

13.In which month more physical classes had taken place?

### Find who had taken more class

In [11]:
WH[['Jikky', 'Kayis', 'Sinny', 'Criji', 'Johnan']].sum()

Jikky     71.2
Kayis     29.6
Sinny     54.5
Criji     43.0
Johnan    23.0
dtype: float64

Jikky had taken more class.

### Find by which mode, most of the time class has taken in month June

In [12]:
tm=pd.crosstab(WH['Month'],WH['Mode of class'])
tm

Mode of class  Online  Physical
Month                          
4                  25         0
5                  24         5
6                  31         6

Online class is mostly taken in the month June.

### Find the sum of Hours of each mentors

In [15]:
WH.groupby('Mode of class').sum()

C:\Users\ALAN PETER\AppData\Local\Temp\ipykernel_40616\1000598899.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  WH.groupby('Mode of class').sum()


Jikky  Kayis  Sinny  Criji  Johnan  Month    year   Day
Mode of class                                                         
Online          57.2   22.6   30.5   23.0    13.5    406  161760  1440
Physical        14.0    7.0   24.0   20.0     9.5     61   22242   195

In [17]:
moc=WH['Mode of class'].value_counts()
moc

Online      80
Physical    11
Name: Mode of class, dtype: int64

The mentors have taken online class than phyical classes.

### Find by which time hour most the class has taken

In [18]:
WH['Time'].mode()

0    9:30 am - 11:30 am
Name: Time, dtype: object

most of the classes had taken in these time hour - 9.30am to 11.30am. `

In [19]:
WH.groupby('Month')['Time'].value_counts()

Month  Time               
4      9:30 am- 10:30 am      12
       11:30 am - 12:30 pm     8
       9:30 am - 11:30 am      3
       10:30 am- 11:30 am      1
       11:30 am -  1:00 pm     1
5      9:30 am - 11:30 am     17
       1:30 pm- 3:30 pm        3
       9:30am-12:00am          2
       9:30am:4:30pm           2
       2: 30PM - 3:50 PM       1
       3: 00PM - 3:30 PM       1
       9:30 am - 11:40 am      1
       9:30am-1 PM             1
       9:30am-3:30pm           1
6      2:30 Pm-4:30pm         14
       11:30 am- 1:30 Pm       9
       9:30 am - 11:30 am      9
       10:00 Am-3:30 Pm        1
       10:00 am - 11:30 am     1
       12:15 pm -12:45 pm      1
       2:30 pm -4:00 pm        1
       9:30am-3:30pm           1
Name: Time, dtype: int64

The mentors mostly took the class from 9.30am to 10.30am in April

The mentors mostly took the class from 9.30am to 11.30am in May.

The mentors mostly took the class from 2.30pm to 4.30pm in June.

In [20]:
WH.groupby('Month')[['Time']].count()

Time
Month      
4        25
5        29
6        37

In [21]:
WH.describe()

Jikky      Kayis      Sinny      Criji     Johnan      Month  \
count  91.000000  91.000000  91.000000  91.000000  91.000000  91.000000   
mean    0.782418   0.325275   0.598901   0.472527   0.252747   5.131868   
std     1.116701   0.888393   1.236527   1.184067   0.940962   0.819333   
min     0.000000   0.000000   0.000000   0.000000   0.000000   4.000000   
25%     0.000000   0.000000   0.000000   0.000000   0.000000   4.000000   
50%     0.000000   0.000000   0.000000   0.000000   0.000000   5.000000   
75%     2.000000   0.000000   0.500000   0.000000   0.000000   6.000000   
max     7.000000   7.000000   7.000000   7.000000   7.000000   6.000000   

         year        Day  
count    91.0  91.000000  
mean   2022.0  17.967033  
std       0.0   8.430699  
min    2022.0   1.000000  
25%    2022.0  11.500000  
50%    2022.0  20.000000  
75%    2022.0  24.500000  
max    2022.0  31.000000

### Find how many physical classes has taken by each mentors through out each month

In [22]:
WH.groupby('Mode of class')[['Jikky', 'Kayis', 'Sinny', 'Criji', 'Johnan']].sum()

Jikky  Kayis  Sinny  Criji  Johnan
Mode of class                                    
Online          57.2   22.6   30.5   23.0    13.5
Physical        14.0    7.0   24.0   20.0     9.5

In these past 3 months most of the online classes had taken by Jikky

and Sinny mostly took the physical classes

In [26]:
WH.groupby('Month')[['Jikky', 'Kayis', 'Sinny', 'Criji', 'Johnan']].sum()

Jikky  Kayis  Sinny  Criji  Johnan
Month                                    
4       11.5   14.5    9.5    1.0     0.0
5       22.7   15.1   22.0   21.0    23.0
6       37.0    0.0   23.0   21.0     0.0

Kayis had taken more class hours in April

Johnan had taken more class hours in May

Jikky had taken more class hours in June

In [28]:
jm=pd.crosstab(WH['Johnan'],WH['Mode of class']).sum()
jm

Mode of class
Online      80
Physical    11
dtype: int64

In [29]:
cm=pd.crosstab(WH['Criji'],WH['Mode of class']).sum()
cm

Mode of class
Online      80
Physical    11
dtype: int64

In [30]:
sm=pd.crosstab(WH['Sinny'],WH['Mode of class']).sum()
sm

Mode of class
Online      80
Physical    11
dtype: int64

In [31]:
km=pd.crosstab(WH['Kayis'],WH['Mode of class']).sum()
km

Mode of class
Online      80
Physical    11
dtype: int64

In [33]:
WH.groupby('Month')['Mode of class'].value_counts()

Month  Mode of class
4      Online           25
5      Online           24
       Physical          5
6      Online           31
       Physical          6
Name: Mode of class, dtype: int64

The most of Online class is taken in June

The most of Physical classes is taken in June

In [34]:
WH.groupby('Month')[['Mode of class']].count()

Mode of class
Month               
4                 25
5                 29
6                 37

### If the the payment is 100 for one hour , then how much salary

In [35]:
WH['Salary_1_jikky']=(WH['Jikky']*100)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  
0        Online      4  2022   11           100.0  
1        Online      4  2022   11           100.0

In [36]:
WH['Salary_1_kayis']=(WH['Kayis']*100)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_kayis  
0        Online      4  2022   11           100.0           100.0  
1        Online      4  2022   11           100.0           100.0

In [37]:
WH['Salary_1_sinny']=(WH['Sinny']*100)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_kayis  
0        Online      4  2022   11           100.0             0.0  
1        Online      4  2022   11           100.0             0.0

In [38]:
WH['Salary_1_criji']=(WH['Criji']*100)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_kayis  
0        Online      4  2022   11           100.0             0.0  
1        Online      4  2022   11           100.0             0.0

In [40]:
WH['Salary_1_Johnan']=(WH['Johnan']*100)
WH.head(2)                       

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_kayis  \
0        Online      4  2022   11           100.0             0.0   
1        Online      4  2022   11           100.0             0.0   

   Salary_1_Johnan  
0              0.0  
1              0.0

### If the the payment for one hour is 150 means how much salary did they got in each month

In [45]:
WH['Salary_2_Johnan']=(WH['Johnan']*150)
WH.head(2)                       

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_kayis  \
0        Online      4  2022   11           100.0           150.0   
1        Online      4  2022   11           100.0           150.0   

   Salary_1_Johnan  Salary_1_sinny  Salary_1_criji  Salary_2_Johnan  
0              0.0             0.0             0.0              0.0  
1              0.0             0.0             0.0              0.0

In [46]:
WH['Salary_2_kayis']=(WH['Kayis']*150)
WH.head(2)                       

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_kayis  \
0        Online      4  2022   11           100.0           150.0   
1        Online      4  2022   11           100.0           150.0   

   Salary_1_Johnan  Salary_1_sinny  Salary_1_criji  Salary_2_Johnan  \
0              0.0             0.0             0.0              0.0   
1              0.0             0.0             0.0              0.0   

   Salary_2_kayis  
0           150.0  
1           150.0

In [47]:
WH['Salary_2_sinny']=(WH['Sinny']*150)
WH.head(2)                       

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_kayis  \
0        Online      4  2022   11           100.0           150.0   
1        Online      4  2022   11           100.0           150.0   

   Salary_1_Johnan  Salary_1_sinny  Salary_1_criji  Salary_2_Johnan  \
0              0.0             0.0             0.0              0.0   
1              0.0             0.0             0.0              0.0   

   Salary_2_kayis  Salary_2_sinny  
0           150.0             0.0  
1           150.0             0.0

In [48]:
WH['Salary_2_criji']=(WH['Criji']*150)
WH.head(2)                       

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_kayis  \
0        Online      4  2022   11           100.0           150.0   
1        Online      4  2022   11           100.0           150.0   

   Salary_1_Johnan  Salary_1_sinny  Salary_1_criji  Salary_2_Johnan  \
0              0.0             0.0             0.0              0.0   
1              0.0             0.0             0.0              0.0   

   Salary_2_kayis  Salary_2_sinny  Salary_2_criji  
0           150.0             0.0             0.0  
1           150.0             0.0             0.0

### Create a dataframe using the data you got with salary of each month of each mentors

In [50]:
WH[['Salary_1_jikky','Salary_1_kayis','Salary_1_sinny','Salary_1_criji','Salary_1_Johnan']].sum()

Salary_1_jikky     7120.0
Salary_1_kayis     4440.0
Salary_1_sinny     8175.0
Salary_1_criji     6450.0
Salary_1_Johnan    3450.0
dtype: float64

In [49]:
WH.groupby('Month')[['Salary_1_jikky','Salary_1_kayis','Salary_1_sinny','Salary_1_criji','Salary_1_Johnan']].sum()

Salary_1_jikky  Salary_1_kayis  Salary_1_sinny  Salary_1_criji  \
Month                                                                   
4              1150.0          2175.0          1425.0           150.0   
5              2270.0          2265.0          3300.0          3150.0   
6              3700.0             0.0          3450.0          3150.0   

       Salary_1_Johnan  
Month                   
4                  0.0  
5               3450.0  
6                  0.0

If per hour they get 100 means;

Kayis earned the most salary in April

Johnan earned a higher salary in May

Jikky earned the most salary in June

# Conclusion

91 rows and 8 columns contained in this table.

Fill null is cleaned that it replaced 0 from nan

The date split into month, day and year as they added in each column.

Jikky took more class at the total of 71.2

The mentors have taken online class than physical.

Jikky mostly took the online class

Sinny mostly took the physical class

Kayis has took more hours of working in April, Kayis earned the most salary in April

Johnan has took more hours of working in May, Johnan earned a higher salary in May

Jikky has mostly took more hours of working in June, Jikky earned the most salary in June

Jikky is working more time.

More physical classes had taken in June.

The mentors mostly took the class from 9.30am to 10.30am in April

The mentors mostly took the class from 9.30am to 11.30am in May.

The mentors mostly took the class from 2.30pm to 4.30pm in June